codec/src/main/java/io/netty/handler/codec/DateFormatter.java

```diff
@@ -171,7 +171,7 @@ public void reset() {
         month = -1;
         yearFound = false;
         year = -1;
-        cal.set(Calendar.MILLISECOND, 0);
+        cal.clear();
```

Comment:
with `set(Calendar.MILLISECOND, 0)`:

```
Benchmark                       Mode  Cnt         Score        Error  Units
formatHttpHeaderDateFormat     thrpt   20   5902113,090 ± 582990,460  ops/s
formatHttpHeaderDateFormatter  thrpt   20  18746747,187 ± 881195,600  ops/s
clientCookieDecoder (old)      thrpt   20    382472,406 ±  15152,519  ops/s
clientCookieDecoder (new)      thrpt   20   3521841,227 ± 266471,188  ops/s
parseHttpHeaderDateFormat      thrpt   20    449382,263 ±  19246,001  ops/s
parseHttpHeaderDateFormatter   thrpt   20   8281767,124 ± 954152,352  ops/s
```

with `clear()`:

```
Benchmark                       Mode  Cnt         Score         Error  Units
formatHttpHeaderDateFormat     thrpt   20   6119849,950 ±  496602,256  ops/s
formatHttpHeaderDateFormatter  thrpt   20  16766570,826 ± 1396209,423  ops/s
clientCookieDecoder (old)      thrpt   20    405931,633 ±   19880,255  ops/s
clientCookieDecoder (new)      thrpt   20   3774983,249 ±  173228,422  ops/s
parseHttpHeaderDateFormat      thrpt   20    449639,477 ±   15248,883  ops/s
parseHttpHeaderDateFormatter   thrpt   20   9890669,831 ±  511413,232  ops/s
```

Perf are about the same, so I'd rather go with clear.